sql/2025/fonts/design/styles_family.sql

```diff
@@ -0,0 +1,73 @@
+-- Section: Design
+-- Question: Which families are popular in CSS?
+-- Normalization: Pages
+
+-- INCLUDE https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/sql/{year}/fonts/common.sql
+
+CREATE TEMPORARY FUNCTION FAMILIES(json STRING)
+RETURNS ARRAY<STRING>
+LANGUAGE js
+OPTIONS (library = ["gs://httparchive/lib/css-font-parser.js", "gs://httparchive/lib/css-utils.js"])
+AS '''
+try {
+  const $ = JSON.parse(json);
+  const result = [];
+  walkDeclarations($, (declaration) => {
+    result.push(parseFontFamilyProperty(declaration.value)[0]);
+  }, {
+    properties: 'font-family',
+    rules: (rule) => rule.type.toLowerCase() === 'font-face'
+  });
+  return result;
+} catch (e) {
+  return [];
+}
+''';
+
+WITH
+families AS (
+  SELECT
+    client,
+    FAMILY_INNER(family) AS family,
+    COUNT(DISTINCT page) AS count,
+    ROW_NUMBER() OVER (PARTITION BY client ORDER BY COUNT(DISTINCT page) DESC) AS rank
+  FROM
+    `httparchive.crawl.parsed_css`,
+    UNNEST(FAMILIES(TO_JSON_STRING(css))) AS family
```

Comment:
```suggestion
CREATE TEMPORARY FUNCTION FAMILIES($ JSON)
RETURNS ARRAY<STRING>
LANGUAGE js
OPTIONS (library = ["gs://httparchive/lib/css-font-parser.js", "gs://httparchive/lib/css-utils.js"])
AS '''
try {
  const result = [];
  walkDeclarations($, (declaration) => {
    result.push(parseFontFamilyProperty(declaration.value)[0]);
  }, {
    properties: 'font-family',
    rules: (rule) => rule.type.toLowerCase() === 'font-face'
  });
  return result;
} catch (e) {
  return [];
}
''';

WITH
families AS (
  SELECT
    client,
    FAMILY_INNER(family) AS family,
    COUNT(DISTINCT page) AS count,
    ROW_NUMBER() OVER (PARTITION BY client ORDER BY COUNT(DISTINCT page) DESC) AS rank
  FROM
    `httparchive.crawl.parsed_css`,
    UNNEST(FAMILIES(css)) AS family
```

Reply:
Uh, interesting. I did not realize custom functions support JSON, too. Thank you for the note! I need to updates at a few places.